### PR TITLE
tket emulator feature additions

### PR DESCRIPTION
**Pull request by ZachMassa**
Please leave all links as is and do not change links to direct links to pdfs. Documents are periodically updated and this will break links if direct links are used.

### CODE REVIEW

**Comment by ZachMassa** [comment]
@microsoft-github-policy-service agree company="Quantinuum"

**Comment by Court72** [comment]
@bradben

Can you review the proposed changes? 
- Note that the validation warnings must be resolved before merging. 
- The changes should also be moved into the private repo for staging and automated checks. Let us know if you'd like help with that.

When the changes are ready for publication, add a `#sign-off` comment to signal that the PR is ready for the review team to merge. 

#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

File: articles/provider-quantinuum.md [hunk 1]
```diff
@@ -94,9 +95,269 @@ Both System Model H1 hardware H1-1 and H1-2 are continuously upgraded throughout
...
+
+Mid-circuit Measurement and Qubit Reuse (MCMR) enables users to measure qubits in the middle of a circuit and re-use them.
+
+Due to the internal level structure of trapped-ion qubits, a mid-circuit measurement may leave the qubit in a non-computational state. All mid-circuit measurements should be followed by initialization if the qubit is to be used again in that circuit. The code examples below demonstrate this.
+
+When a subset of qubits is measured in the middle of the circuit, the classical information from these measurements can be used to condition future elements of the circuit. The examples also highlight this usage. 
+
+For information on Mid-circuit Measurement and Qubit Reuse (MCMR) in Quantinuum systems, see the *System Model H1 Product Data Sheet* on the [System Model H1] page.
+
+#### [MCMR with Q# Provider](##tab/tabid-mcmr-with-q-provider)
```
Comment (Bradben) [inline articles/provider-quantinuum.md:121]:
```suggestion
#### [MCMR with Q# Provider](#tab/tabid-mcmr-with-q-provider)
```

File: articles/provider-quantinuum.md [hunk 2]
```diff
@@ -94,9 +95,269 @@ Both System Model H1 hardware H1-1 and H1-2 are continuously upgraded throughout
...
+    }
+    CNOT(qubits[0], qubits[1]);
+
+    # Measure qubits and return results
+    let res2 = Measure([PauliZ, PauliZ], qubits);
+    return [res1, res2];
+}
+```
+
+#### [MCMR with Qiskit Provider](##tab/tabid-mcmr-with-qiskit-provider)
```
Comment (Bradben) [inline articles/provider-quantinuum.md:153]:
```suggestion
#### [MCMR with Qiskit Provider](#tab/tabid-mcmr-with-qiskit-provider)
```

File: articles/provider-quantinuum.md [hunk 3]
```diff
@@ -94,9 +95,269 @@ Both System Model H1 hardware H1-1 and H1-2 are continuously upgraded throughout
...
+# Print out the circuit
+circuit.draw()
+```
+
+### Arbitrary Angle ZZ Gates
+
+Quantinuum's native gate set includes arbitrary angle ZZ gates. This is beneficial for reducing the 2-qubit gate count for many quantum algorithms and gate sequences. For information on Arbitrary Angle ZZ gates in Quantinuum systems, see the *System Model H1 Product Data Sheet* on the [System Model H1] page.
 
-Technical details for the System Model H1 and System Model H1 Emulators can be found in Quantinuum's product data sheets on the [System Model H1](https://www.quantinuum.com/products/h1) page alongside links to Quantinuum data repositories and how to cite usage of Quantinuum systems.
+#### [Arbitrary Angle ZZ Gates with Q# Provider](##tab/tabid-arbitrary-angle-zz-gates-with-q-provider)
```
Comment (Bradben) [inline articles/provider-quantinuum.md:189]:
```suggestion
#### [Arbitrary Angle ZZ Gates with Q# Provider](#tab/tabid-arbitrary-angle-zz-gates-with-q-provider)
```

File: articles/provider-quantinuum.md [hunk 4]
```diff
@@ -94,9 +95,269 @@ Both System Model H1 hardware H1-1 and H1-2 are continuously upgraded throughout
...
+    # Measure qubits and return results
+    for i in IndexRange(qubits) {
+        set resultArray w/= i <- M(qubits[i]);
+    }
+    
+    return resultArray;
+}
+```
+
+#### [Arbitrary Angle ZZ Gates with Qiskit Provider](##tab/tabid-arbitrary-angle-zz-gates-with-qiskit-provider)
```
Comment (Bradben) [inline articles/provider-quantinuum.md:219]:
```suggestion
#### [Arbitrary Angle ZZ Gates with Qiskit Provider](#tab/tabid-arbitrary-angle-zz-gates-with-qiskit-provider)
```

File: articles/provider-quantinuum.md [hunk 5]
```diff
@@ -94,9 +95,269 @@ Both System Model H1 hardware H1-1 and H1-2 are continuously upgraded throughout
...
+circuit.measure_all()
+```
+
+### Emulator Noise Parameters
+
+Users have the option of experimenting with the noise parameters of the Quantinuum emulators. **Only a few of the available noise parameters are highlighted** here demonstrating how to pass through the parameters in the Azure Quantum providers.
+
+For more information on the full set of noise parameters available, see the *System Model H1 Emulator Product Data Sheet* on the [System Model H1] page.
+
+#### [Emulator Noise Parameters with Q# Provider](##tab/tabid-emulator-noise-parameters-with-q-provider)
```
Comment (Bradben) [inline articles/provider-quantinuum.md:249]:
```suggestion
#### [Emulator Noise Parameters with Q# Provider](#tab/tabid-emulator-noise-parameters-with-q-provider)
```

File: articles/provider-quantinuum.md [hunk 6]
```diff
@@ -94,9 +95,269 @@ Both System Model H1 hardware H1-1 and H1-2 are continuously upgraded throughout
...
+
+# Pass in the options when submitting the job
+result = qsharp.azure.execute(GenerateRandomBit, 
+                              shots = 100, 
+                              jobName = "Experiment with Emulator Noise Parameters", 
+                              timeout = 240,
+                              jobParams = option_params)
+```
+
+#### [Emulator Noise Parameters with Qiskit Provider](##tab/tabid-emulator-noise-parameters-with-qiskit-provider)
```
Comment (Bradben) [inline articles/provider-quantinuum.md:279]:
```suggestion
#### [Emulator Noise Parameters with Qiskit Provider](#tab/tabid-emulator-noise-parameters-with-qiskit-provider)
```

File: articles/provider-quantinuum.md [hunk 7]
```diff
@@ -94,9 +95,269 @@ Both System Model H1 hardware H1-1 and H1-2 are continuously upgraded throughout
...
+In the H-Series software stack, the optimization level applied is set with the `tket-opt-level` parameter. *The default compilation setting for all circuits submitted to H-Series systems is optimization level 2.*
+
+Users who would like to experiment with the TKET compilation passes and see what optimizations would apply to their circuits *before* submitting any jobs can see the *Quantinuum_compile_without_api.ipynb* notebook in the [`pytket-quantinuum` Examples] folder.
+
+For more information on `pytket`, see the following links:
+
+- [`pytket`]
+- [`pytket` User Manual]
+
+#### [TKET Compilation with Q# Provider](##tab/tabid-tket-compilation-with-q-provider)
```
Comment (Bradben) [inline articles/provider-quantinuum.md:322]:
```suggestion
#### [TKET Compilation with Q# Provider](#tab/tabid-tket-compilation-with-q-provider)
```

File: articles/provider-quantinuum.md [hunk 8]
```diff
@@ -94,9 +95,269 @@ Both System Model H1 hardware H1-1 and H1-2 are continuously upgraded throughout
...
+
+# Pass in the options when submitting the job
+result = qsharp.azure.execute(GenerateRandomBit, 
+                              shots = 100, 
+                              jobName = "Experiment with TKET Compilation", 
+                              timeout = 240,
+                              jobParams = option_params)
+```
+
+#### [TKET Compilation with Qiskit Provider](##tab/tabid-tket-compilation-with-qiskit-provider)
```
Comment (Bradben) [inline articles/provider-quantinuum.md:341]:
```suggestion
#### [TKET Compilation with Qiskit Provider](#tab/tabid-tket-compilation-with-qiskit-provider)
```

File: articles/provider-quantinuum.md [hunk 9]
```diff
@@ -137,7 +398,20 @@ where:
...
+Quotas are based on plan selection and can be increased with a support ticket. To see your current limits and quotas, go to the **Credits and quotas** blade and select the **Quotas** tab of your workspace on the [Azure portal]. For more information, see [Azure Quantum quotas](xref:microsoft.quantum.quotas).
 
 > [!NOTE]
 > If you are using an [Azure Quantum Credits](xref:microsoft.quantum.credits) plan, and not a billing plan, the quotas information maps to your allocated credits. In that case, the quota lists the total number of credits you have received.
+
+***
+
+[Quantinuum]: https://www.quantinuum.com
+[System Model H1]: https://www.quantinuum.com/products/h1
+[`MResetZ`]: https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.measurement.mresetz
```
Comment (Bradben) [inline articles/provider-quantinuum.md:410]:
```suggestion
[`MResetZ`]: /qsharp/api/qsharp/microsoft.quantum.measurement.mresetz
```

**Review by ffrachon — COMMENTED** [review]
- You have "Powered by Honeywell". Change to "powered by Quantinuum"
- Add hyperlink to "To see Quantinuum's billing plans, visit Azure Quantum pricing": https://learn.microsoft.com/azure/quantum/pricing

**Comment by ZachMassa** [comment]
> * You have "Powered by Honeywell". Change to "powered by Quantinuum"

This needs to remain as "Powered by Honeywell" as part of our on going agreement with Honeywell.

**Comment by ZachMassa** [comment]
> * Add hyperlink to "To see Quantinuum's billing plans, visit Azure Quantum pricing": https://learn.microsoft.com/azure/quantum/pricing

@ffrachon 

I have updated this link. It is still not displaying in GitHub properly but we believe this is due to the connectors not being setup in GitHub. 

image

**Comment by Bradben** [comment]
@ZachMassa - FYI, I'm actually tracking these changes in another PR in our private repo, where it gives me a preview, more extensive verification checks, etc. All your updates look fine so far, thanks!

**Pull request closed without merging**